### PR TITLE
1040 Add tests for emailSchema

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#1040

### Dependencies

none

### Description

Add tests for emailSchema - [context](https://metriport.slack.com/archives/C04T256DQPQ/p1721345749355369?thread_ts=1717952771.261659&cid=C04T256DQPQ)

### Testing

- Local
  - [x] unit tests
- Staging
  - none
- Sandbox
  - none
- Production
  - none

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/api-sdk/src/medical/models/__tests__/demographics.test.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    it("accepts null email", async () => {
+      const contact = makeContact();
+      contact.email = null;
+      expect(() => contactSchema.parse(contact)).not.toThrow();
+    });
+
+    it("accepts undefined email", async () => {
+      const contact = makeContact();
+      contact.email = null;
```
Comment (RamilGaripov) [inline packages/api-sdk/src/medical/models/__tests__/demographics.test.ts:19]: `= undefined`

File: packages/api-sdk/src/medical/models/__tests__/demographics.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    });
+
+    it("accepts email with dot", async () => {
+      expect(() => emailSchema.parse("test.two@metriport.com")).not.toThrow();
+    });
+
+    it("fails when gets email with colon", async () => {
+      expect(() => emailSchema.parse("test:two@metriport.com")).toThrow();
+    });
+  });
```
Comment (RamilGaripov) [inline packages/api-sdk/src/medical/models/__tests__/demographics.test.ts:44]: we don't wanna check more cases? email without `@`? email with subdomain, etc?

**Review by RamilGaripov — CHANGES_REQUESTED** [review]

File: packages/api-sdk/src/medical/models/__tests__/demographics.test.ts (hunk 1 above)
Comment (leite08) [reply packages/api-sdk/src/medical/models/__tests__/demographics.test.ts:19]: good catch!

File: packages/api-sdk/src/medical/models/__tests__/demographics.test.ts (hunk 2 above)
Comment (leite08) [reply packages/api-sdk/src/medical/models/__tests__/demographics.test.ts:44]: yeah, we can def add more cases, but I was not trying to replicate Zod's test, mostly trying to replicate scenarios we've been through and want to know what to expect - and have the code replicating that.

I'll add a few more

Comment (leite08) [reply packages/api-sdk/src/medical/models/__tests__/demographics.test.ts:44]: done

File: packages/api-sdk/src/medical/models/__tests__/demographics.test.ts (hunk 1 above)
Comment (leite08) [reply packages/api-sdk/src/medical/models/__tests__/demographics.test.ts:19]: done

File: packages/api-sdk/src/medical/models/__tests__/demographics.test.ts (hunk 2 above)
Comment (RamilGaripov) [reply packages/api-sdk/src/medical/models/__tests__/demographics.test.ts:57]: 🚀 🚀

**Review by RamilGaripov — APPROVED** [review]
🚀 🚀 🚀 🚀 🚀 🚀 🚀 🚀

**Pull request merged**